torchmetrics/functional/image/uqi.py

```diff
@@ -0,0 +1,245 @@
+    dtype = preds.dtype
+    kernel = _gaussian_kernel(channel, kernel_size, sigma, dtype, device)
+    pad_h = (kernel_size[0] - 1) // 2
+    pad_w = (kernel_size[1] - 1) // 2
+
+    preds = F.pad(preds, (pad_h, pad_h, pad_w, pad_w), mode="reflect")
+    target = F.pad(target, (pad_h, pad_h, pad_w, pad_w), mode="reflect")
+
+    input_list = torch.cat((preds, target, preds * preds, target * target, preds * target))  # (5 * B, C, H, W)
+    outputs = F.conv2d(input_list, kernel, groups=channel)
+    output_list = outputs.split(preds.shape[0])
+
+    mu_pred_sq = output_list[0].pow(2)
+    mu_target_sq = output_list[1].pow(2)
+    mu_pred_target = output_list[0] * output_list[1]
+
+    sigma_pred_sq = output_list[2] - mu_pred_sq
+    sigma_target_sq = output_list[3] - mu_target_sq
+    sigma_pred_target = output_list[4] - mu_pred_target
+
+    upper = 2 * sigma_pred_target
+    lower = sigma_pred_sq + sigma_target_sq
+
+    uqi_idx = ((2 * mu_pred_target) * upper) / ((mu_pred_sq + mu_target_sq) * lower)
+    uqi_idx = uqi_idx[..., pad_h:-pad_h, pad_w:-pad_w]
+
+    return reduce(uqi_idx, reduction)
+
+
+def universal_image_quality_index(
+    preds: Tensor,
+    target: Tensor,
+    kernel_size: Sequence[int] = (11, 11),
+    sigma: Sequence[float] = (1.5, 1.5),
+    reduction: str = "elementwise_mean",
+    data_range: Optional[float] = None,
+) -> Tensor:
+    """Universal Image Quality Index.
+
+    Args:
+        preds: estimated image
+        target: ground truth image
+        kernel_size: size of the gaussian kernel (default: (11, 11))
+        sigma: Standard deviation of the gaussian kernel (default: (1.5, 1.5))
+        reduction: a method to reduce metric score over labels.
+
+            - ``'elementwise_mean'``: takes the mean (default)
+            - ``'sum'``: takes the sum
+            - ``'none'``: no reduction will be applied
+
+        data_range: Range of the image. If ``None``, it is determined from the image (max - min)
+
+    Return:
+        Tensor with UQI score
+
+    Raises:
+        TypeError:
+            If ``preds`` and ``target`` don't have the same data type.
+        ValueError:
+            If ``preds`` and ``target`` don't have ``BxCxHxW shape``.
+        ValueError:
+            If the length of ``kernel_size`` or ``sigma`` is not ``2``.
+        ValueError:
+            If one of the elements of ``kernel_size`` is not an ``odd positive number``.
+        ValueError:
+            If one of the elements of ``sigma`` is not a ``positive number``.
+
+    Example:
+        >>> from torchmetrics.functional import universal_image_quality_index
+        >>> preds = torch.rand([16, 1, 16, 16])
+        >>> target = preds * 0.75
+        >>> universal_image_quality_index(preds, target)
+        tensor(0.9216)
+
+    References:
+    [1] Zhou Wang and A. C. Bovik, "A universal image quality index," in IEEE Signal Processing Letters, vol. 9,
+    no. 3, pp. 81-84, March 2002, doi: 10.1109/97.995823.
```

Comment:
Please add this reference into `docs/source/links.rst`

Reply:
Thank you @stancld, I have fixed this :)